### ISSUE TITLE
re: iTunes Now Playing

### ISSUE THREAD

**Issue by sfai05** [matryer/bitbar-plugins#112]
FAO: (@daturkel)
Its not working when listen to Apple Music. I did some research and find it difficult to do that so can we just hide the widget when listen to Apple Music ?

**Comment by daturkel** [comment]
I'm not sure why this would be the case. Feel free to submit a pull request to toggle the plugin's display when it's using Apple Music.

**Comment by blockloop** [comment]
It works for me using Apple Music using El Capitan.

**Comment by citoki** [comment]
also not working on my machine (10.11.2), get the following error in terminal:
```sh
54:60: execution error: Can’t make name of «class pTrk» of application "iTunes" into type string. (-1700)
56:62: execution error: Can’t make «class pArt» of «class pTrk» of application "iTunes" into type string. (-1700)
55:61: execution error: Can’t make «class pAlb» of «class pTrk» of application "iTunes" into type string. (-1700)
▶ -
---
Track: | color=#333333
Artist: | color=#333333
Album: | color=#333333
---
Pause | bash=/Users/d060015/Sites/github-apps/bitbar-plugins/Enabled/itunes.10s.sh param1=playpause terminal=false
Previous | bash=/Users/d060015/Sites/github-apps/bitbar-plugins/Enabled/itunes.10s.sh param1=previous terminal=false
Next | bash=/Users/d060015/Sites/github-apps/bitbar-plugins/Enabled/itunes.10s.sh param1=next terminal=false
```

![image](https://cloud.githubusercontent.com/assets/7259873/12510897/8071862c-c10d-11e5-9ac4-14578cea6f05.png)

How can I debug to the root cause?

**Comment by blockloop** [comment]
Can you run the following in Terminal and paste the output please?

`osascript -e 'tell application "iTunes" to name of current track'`

**Comment by citoki** [comment]
thanks, here you go:
```sh
$ osascript -e 'tell application "iTunes" to name of current track'
29:33: execution error: iTunes got an error: Can’t get name of current track. (-1728)
```

**Comment by citoki** [comment]
I have to be more precise on that issue. When I listen to a radio station via iTunes tab 'Radio' it works fine. When I choose beats 1 for example, the output is correct:
```sh
$ osascript -e 'tell application "iTunes" to name of current track'
Beats 1
```
The error is only shown when I choose music via 'For, You', 'New', 'Connect' or 'iTunes Store'.

**Comment by alexrockt** [comment]
This is a known bug and was submitted to Apple by multiple people. Can't do anything about it unless they fix it.

**Comment by matryer** [comment]
Closing, as there seems to be nothing we can do.

**Comment by loretoparisi** [comment]
@matryer do you have the radar that has been filed to Apple? It would be helpful to check it something new was done.

**Comment by alexrockt** [comment]
This has been fixed in iTunes 12.4

**Comment by loretoparisi** [comment]
Specifically,

```shell
[ { data: '27,024000167847',
    exitCode: 0,
    cli: 'osascript -e tell application "iTunes" to player position as string' },
  { data: 'Wherever I Go',
    exitCode: 0,
    cli: 'osascript -e tell application "iTunes" to name of current track as string' },
  { data: 'OneRepublic',
    exitCode: 0,
    cli: 'osascript -e tell application "iTunes" to artist of current track as string' },
  { data: 'Wherever I Go - Single',
    exitCode: 0,
    cli: 'osascript -e tell application "iTunes" to album of current track as string' } ]
err data: 39:52: 
```

we have that player position, track name, artist name works properly now from `News` and from `Music` as well. So now all iTunes tabs are covered, `Radio` was working already.
